libs/content/workspace/src/tab/markdown_editor/editor.rs

```diff
@@ -47,10 +48,10 @@ pub struct EditorResponse {
 }
 
 // makes for fewer arguments in a few places
-#[derive(Clone, Copy)]
 pub struct HoverSyntaxRevealDebounceState {
-    pub pointer_offset: Option<DocCharOffset>,
-    pub pointer_offset_updated_at: Instant,
+    pub pointer_ast_leaf_node: Option<usize>,
+    pub updated_at: Instant,
+    pub frame_start: Instant,
```

Comment:
ironically, one troublesome bug was a race condition which I addressed using the technique I wrote in the lockbook document that I was using to demo (in the paragraph visible in the demo!)